dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -55,32 +55,32 @@ public Pattern getParameterPattern() {
 
     public Map<String, DbParameterAccessor> getAllColumns(
             final String tableOrViewName) throws SQLException {
-        String qry = "SELECT COLUMNNAME, COLUMNDATATYPE "
+        String qry = "SELECT COLUMNNAME AS COLUMN_NAME, COLUMNDATATYPE AS TYPE_NAME, "
+                + " ROW_NUMBER() OVER() AS ORDINAL_POSITION, " +
+                + DatabaseMetaData.procedureColumnIn + " AS COLUMN_TYPE "
                 + "FROM SYS.SYSCOLUMNS WHERE REFERENCEID = "
                 + "(SELECT TABLEID FROM SYS.SYSTABLES WHERE TABLENAME = ?)";
-        return readIntoParams(tableOrViewName.toUpperCase(), qry);
+        return fetchIntoParams(tableOrViewName.toUpperCase(), qry);
     }
 
-    private Map<String, DbParameterAccessor> readIntoParams(
-            String tableOrViewName, String query) throws SQLException {
+    private Map<String, DbParameterAccessor> readIntoParams(ResultSet rs)
```

Comment:
`readIntoParams` and `getObjectParameters` are now pretty similar. How can we resolve that code duplication?